src/main/java/io/pravega/connectors/flink/FlinkPravegaWriter.java

```diff
@@ -223,7 +223,7 @@ protected AbstractInternalWriter createInternalWriter() {
 
     @VisibleForTesting
     protected ExecutorService createExecutorService() {
-        return Executors.newFixedThreadPool(5);
+        return Executors.newSingleThreadExecutor();
```

Comment:
One clarifying question. This executor is used with the non-transactional writer with a `whenCompleteAsync`. Can we restructure it to use the default thread pool instead? I'm just wondering if we really need this executor in the first place. Otherwise, the change looks good.

Reply:
I think we would best pull in whomever wrote the original code to get their perspective on that.    And I would prefer we do that later.  For now, I propose we go with this simple improvement.

Reply:
Ok, I have created an issue for it: #157